### ISSUE TITLE
INPUT_FILTER script doesn't work

### ISSUE THREAD

**Issue by viveksood97** [doxygen/doxygen#10211]
**Describe the bug**
I created a script for INPUT_FILTER setting but it does not produce any docs. I have attached the Doxyfile.in and the input_filter script. The script is used to remove visibility macros all of which have _PUBLIC or _LOCAL

[Doxyfile.txt](https://github.com/doxygen/doxygen/files/12261651/Doxyfile.txt)
[input_filter.txt](https://github.com/doxygen/doxygen/files/12261655/input_filter.txt)

**Expected behavior**
Docs get generated

**Screenshots**
If useful, add screenshots to help explain your problem.

**To Reproduce**
N/A

**Version**
v 1.9.1  

**Stack trace**
N/A

**Additional context**
N/A

**Comment by albert-github** [comment]
- an example source would be good to have, so we are always talking about the same
- I looked in the input_filer and I see that you write to an output file but as you can see in the output of your issue #10207"
   ```
   Executing popen(/home/vsood-dev/module/docs/input_filter.py "/home/vsood-dev/module/src/sample_pkg/include/sample_pkg/sample_pkg.hpp")
  ```
  in other words the input_filter  is called with just 1 argument and in the documentation it is stated:
  > Doxygen will then use the output that the filter program writes to standard output

  in other words you have to write to stdout

**Comment by viveksood97** [comment]
I made the change and still no docs. Here's the new file

[input_filter.txt](https://github.com/doxygen/doxygen/files/12262232/input_filter.txt)

I also tried with print() and still no change

Here's the source hpp file
[example_hpp.txt](https://github.com/doxygen/doxygen/files/12262330/example_hpp.txt)

**Comment by albert-github** [comment]
When running the input filter I get the warning:
```
Usage: input_filter.py <input_file> <output_file>
```
this is due to the lines, that have not been adjusted:
```
    if len(sys.argv) != 3:
        print("Usage: input_filter.py <input_file> <output_file>")
```

**Comment by viveksood97** [comment]
That fixed it!. Thanks a ton
